commit 18094661910d9e5706d3ae91ec2695ceaf489b9e
Author: Sergio Ahumada <sergio.ahumada@digia.com>
Date:   Mon Nov 19 16:45:33 2012 +0100

    test: Mark some tst_QHostInfo functions as XFAIL
    
      tst_QHostInfo::cache()
      tst_QHostInfo::abortHostLookup()
    
    Task-number: QTBUG-28079
    Change-Id: I6f524cdee86bc77583852327e18c6d4a79595323
    Reviewed-by: Caroline Chao <caroline.chao@digia.com>

diff --git a/tests/auto/qhostinfo/tst_qhostinfo.cpp b/tests/auto/qhostinfo/tst_qhostinfo.cpp
index b5a1b87..16b731a 100644
--- a/tests/auto/qhostinfo/tst_qhostinfo.cpp
+++ b/tests/auto/qhostinfo/tst_qhostinfo.cpp
@@ -594,6 +594,13 @@ void tst_QHostInfo::cache()
     // loopkup second time, result should come directly
     valid = false;
     result = qt_qhostinfo_lookup("localhost", this, SLOT(resultsReady(QHostInfo)), &valid, &id);
+#if defined(Q_OS_LINUX) || defined(Q_WS_QWS)
+    if (valid != true)
+        QEXPECT_FAIL("", "QTBUG-28079", Continue);
+#endif
+#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
+    QEXPECT_FAIL("", "QTBUG-28079", Continue);
+#endif
     QVERIFY(valid == true);
     QVERIFY(!result.addresses().isEmpty());
 
@@ -631,6 +638,13 @@ void tst_QHostInfo::abortHostLookup()
     //it is assumed that the DNS request/response in the backend is slower than it takes to call abort
     QHostInfo::abortHostLookup(id);
     QTestEventLoop::instance().enterLoop(5);
+#if defined(Q_OS_LINUX) || defined(Q_WS_QWS)
+    if (lookupsDoneCounter != 0)
+        QEXPECT_FAIL("", "QTBUG-28079", Continue);
+#endif
+#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
+    QEXPECT_FAIL("", "QTBUG-28079", Continue);
+#endif
     QCOMPARE(lookupsDoneCounter, 0);
 }
 
